src/libpmem2/x86_64/memcpy/memcpy_nt_avx.c

```diff
@@ -154,14 +155,29 @@ memmove_movnt_avx_fw(char *dest, const char *src, size_t len, flush_fn flush)
 		len -= cnt;
 	}
 
-	while (len >= 8 * 64) {
+	const char *srcend = src + len;
+	prefetch_ini_fw(src, len);
+
+	while (len >= PERF_BARRIER_SIZE) {
+		prefetch_next_fw(src, srcend);
```

Comment:
will this prefetch the same thing twice?
Maybe move this to the end of the loop.

I kinda recall you mentioning that, but now looking at the code I think we should at least add a comment stating that this is on purpose.

Reply:
Nope, it won't. 

prefetch_ini_fw prefetches 4k at offset 0.
first prefetch_next_fw prefetches 768 bytes at offset 4k
second prefetch_next_fw prefetches 768 bytes at offset 4k+768
N-th prefetch_next_fw prefetches 768 bytes at offset 4k+768*(N-1).